worker/codalabworker/worker.py

```diff
@@ -92,13 +97,30 @@ def run(self):
         if not self.shared_file_system:
             self._dependency_manager.start_cleanup_thread()
 
+        self._worker_state_manager.resume_previous_runs(
+                lambda run_info: Run.deserialize(
+                    self._bundle_service, self._docker, self._image_manager, self, run_info)
+        )
+        with self._resource_lock:
+            run_sets = self._worker_state_manager.map_runs(lambda run: (run._cpuset, run._gpuset))
+            for cpuset, gpuset in run_sets:
```

Comment:
Please document what the high-level logic here is

Reply:
ok